airbyte-integrations/connectors/source-freshdesk/source_freshdesk/api.py

```diff
@@ -266,6 +266,56 @@ def list(self, fields: Sequence[str] = None) -> Iterator[dict]:
         params = {"include": "description"}
         yield from self.read(partial(self._api_get, url="tickets"), params=params)
 
+    # This block extends TicketsAPI Stream to overcome '300 page' server error.
+    # Since the TicketsAPI Stream list has a 300 page pagination limit, after 300 pages, update the parameters with
+    # query using 'updated_since' = last_record, if there is more data remaining.
+    def get_tickets(self, getter: Callable, params: Mapping[str, Any] = None) -> Iterator:
+        """Read using getter"""
+        params = params or {}
+        # the maximum page allowed to pull during pagination.
+        ticket_paginate_limit = 300
+        # Start page
+        page = 1
+        # Initial request parameters
+        params = {**params, "per_page": self.result_return_limit}
+
+        while True:
+            params["page"] = page
+            batch = list(getter(params=params))
+            yield from batch
+
+            if len(batch) < self.result_return_limit:
+                return False
```

Comment:
```suggestion
                return iter(())
```

Reply:
Less confusing to return an empty iterator than to return a falsy iterator value.